definitions/features/container_gateway_database.rb

```diff
@@ -0,0 +1,40 @@
+class Features::ContainerGatewayDatabase < ForemanMaintain::Feature
+  CONTAINER_GATEWAY_DB_CONFIG = '/etc/foreman-proxy/settings.d/container_gateway.yml'.freeze
+
+  include ForemanMaintain::Concerns::BaseDatabase
+  include ForemanMaintain::Concerns::DirectoryMarker
+
+  metadata do
+    label :container_gateway_database
+
+    confine do
+      file_nonzero?(CONTAINER_GATEWAY_DB_CONFIG)
+    end
+  end
+
+  def configuration
+    @configuration || load_configuration
+  end
+
+  def services
+    [
+      system_service('postgresql', 10, :component => 'container_gateway',
+        :db_feature => feature(:container_gateway_database)),
+    ]
+  end
+
+  private
+
+  def load_configuration
+    config = YAML.load(File.read(CONTAINER_GATEWAY_DB_CONFIG))
+    @configuration = {}
+    connection_string = config[:db_connection_string]
+    if connection_string
+      uri = URI.parse(connection_string)
+      @configuration['connection_string'] = connection_string
+      @configuration['user'] = 'foreman-proxy'
```

Comment:
can't the user configure a different db user, and this one is just the fallback?

Reply:
oh wait, no, reading further below, that's the "shell user" not "database user"? I'm confused :D

Reply:
It is indeed a shell user.  The container gateway database (and maybe more in the future) does not configure any password by default since it uses unix-domain socket authentication.  As such, `root` can't (as far as I can tell) log into the database in any way, so I've created a generic way to set the shell user for doing database commands.